src/nvim/screen.c

```diff
@@ -452,6 +452,25 @@ void update_screen(int type)
 
 }
 
+/// Prepare for updating one or more windows.
+/// Caller must check for "updating_screen" already set to avoid recursiveness.
+static void update_prepare(void)
+{
+  updating_screen = true;
+  start_search_hl();
```

Comment:
Could pass `bool` to update_prepare to make `start_search_hl` optional. Same applies to `update_finish` for `end_search_hl`.

Reply:
May need 8.1.0192.